addons/repair/models/stock_picking.py

```diff
@@ -79,6 +77,21 @@ def _compute_is_repairable(self):
             if not(picking_type.code == 'incoming' and picking_type.return_type_of_ids):
                 picking_type.is_repairable = False
 
+    @api.onchange('code')
+    def _onchange_picking_code(self):
+        super()._onchange_picking_code()
+
+        if self.code == 'repair_operation':
+            warehouse = self.env['stock.warehouse'].search([('company_id', '=', self.company_id.id)], limit=1)
+            stock_location = warehouse.lot_stock_id
+            prod_location = self.env['stock.location'].search([('usage', '=', 'production'), ('company_id', '=', self.company_id.id)], limit=1)
+            scrap_location = self.env['stock.location'].search([('scrap_location', '=', True), ('company_id', 'in', [self.company_id.id, False])], limit=1)
+
+            self.default_location_src_id = stock_location.id
+            self.default_location_dest_id = prod_location.id
+            self.default_remove_location_dest_id = scrap_location.id
+            self.default_recycle_location_dest_id = stock_location.id
+
```

Comment:
You should not use compute in order to keep the same behavior backend/frontend. Also rco made a fix on the model. I will explain you wed

Reply:
Done like that in stock module. To update in master